File: SINGLY LINKED LIST/DLinkedList.java
public class DLinkedList {
    class Node{
        int data;
        Node prev;
        Node next;

        Node(int data){
            this.data = data;
        }
    }

    public Node head = null;
    public Node tail = null;

    public void addNode(int data){
        Node newNode = new Node(data);
        if(head == null){
            head = newNode;
        }
        else{
            tail.next = newNode;
            newNode.prev = tail;
        }
        tail = newNode;
    }

    public void delete(int data){
        Node temp = head;
       
        
        if(temp != null && temp.data == data){
           head = temp.next;
           head.prev = null;
           return;
        }
        while(temp != null && temp.data != data){
           
            temp = temp.next;  
            
        }
        if(temp == tail){
            tail = temp.prev;
            tail.next = null;
            return;
        }
       
        temp.prev.next = temp.next;
        temp.next.prev = temp.prev;
        
        
    }

    public void insertAfter(int nextTo , int data){
        Node newnNode = new Node(data);
        Node temp = head;
        while( temp !=null && temp.data != nextTo){
            temp = temp.next;
        }
        newnNode.prev = temp.next.prev;
        temp.next.prev = newnNode;
        newnNode.next = temp.next;
        temp.next = newnNode;
        

    }

    public void display(){
        Node temp = head;
        
        while ( temp != null ){
            System.out.println(temp.prev+"  "+temp.data +"  "+temp.next);
            temp = temp.next;
        }
    }

    public static void main(String[] args) {
        DLinkedList list = new DLinkedList();
        list.addNode(20);
        list.addNode(40);
        list.addNode(60);
        list.addNode(80);
        list.addNode(100);

        list.insertAfter(60, 70);
       
        list.display();
    }
}

File: DOUBLY LINKED LIST/dLinkedList.js
class Node {
    constructor(data){
        this.value = data
        this.prev = null
        this.next = null
    }
}
class DLL{
    constructor(){
        this.head = null
        this.tail = null
    }

    addNode (data){
        const newNode = new Node(data)
        if(this.head == null){
            this.head = newNode
        }
        else{
            this.tail.next = newNode
            newNode.prev = this.tail
        }
        this.tail = newNode
    }

    display(){
        let temp = this.head
        while(temp != null){
            console.log(temp.value,'->');
            temp = temp.next
        }
    }
}

const List = new DLL()
List.addNode(10)
List.addNode(20)
List.addNode(30)
List.addNode(40)
List.addNode(50)
List.addNode(60)
List.display()
